Library/Homebrew/cmd/configure-shell.rb

```diff
@@ -0,0 +1,39 @@
+#:  * `configure-shell`:
+#:    Configure the shell to include Homebrew into your PATH, MANPATH, and INFOPATH.
+#:    An environment file will be created that will be sourced in your ~/.profile
+#:    (or your .bash_profile or .zprofile - only existing files will be modified)
+
+module Homebrew
+  module_function
+
+  def configure_shell
+    # Write environment details (PATH, MANPATH, and INFOPATH) to a file that will export them
+
+    # Ensure the directory for this file exists, and then write the file
+    FileUtils.mkdir_p HOMEBREW_PREFIX/"etc"
+
+    # Contents for the environment file
+    environment = <<~EOS
+      # Homebrew environment
+      # Created by running brew configure-shell
+      export PATH="#{HOMEBREW_PREFIX}/bin:$PATH"
+      export MANPATH="#{HOMEBREW_PREFIX}/share/man:$MANPATH"
+      export INFOPATH="#{HOMEBREW_PREFIX}/share/info:$INFOPATH"
+    EOS
+
+    # If the file already exists,
+    env_path = HOMEBREW_PREFIX/"etc/environment"
+    File.write env_path, environment unless File.file?(env_path)
+    odie "Error: #{env_path} already exists. If you wish to replace this file, remove this file and run brew configure-shell again." if (File.read env_path) != environment
```

Comment:
I agree - previously the linter yelled at me for writing `if` statements that weren't on the end of a line, so that's why I made that line so long. But somehow it's not yelling now (I guess I had to put `if` at the end of some *other* line).